src/datumaro/plugins/transforms.py

```diff
@@ -1967,12 +1981,22 @@ def transform_item(self, item):
                 "Item %s: TableRow info is required for this " "transform" % (item.id,)
             )
 
+        sep_token = self._sep_token
         refined_media = self.refine_tabular_media(item) if item.media.has_data else None
-        annotation_values = {
-            key: refined_media.data[key] for key in item.annotations[0].values.keys()
-        }  # only for tabular
-        refined_annotations = [
-            ann.wrap(values=annotation_values) if isinstance(ann, Tabular) else ann
-            for ann in item.annotations
-        ]
+        refined_annotations = []
+        for ann in item.annotations:
+            if isinstance(ann, Tabular):
+                annotation_values = {
+                    key: refined_media.data[key] for key in item.annotations[0].values.keys()
+                }  # only for tabular
+                ann = ann.wrap(values=annotation_values)
+            elif isinstance(ann, Caption):
+                value = [
+                    f"{key}{sep_token}{refined_media.data[key]}"
+                    for key in refined_media.data.keys()
+                    if ann.caption.startswith(key)
+                ]
+                ann = ann.wrap(caption=value[0])
```

Comment:
Is it guaranteed that `value` has always one element?

Reply:
As we get value through the above lines, caption would have only one value.

Reply:
`value` comes from this for-loop `for key in refined_media.data.keys()`. How did you guarantee it is singular?

Reply:
So far, there is an expectation to do `AstypeAnnotations` for a tabular dataset before `Clean` transformation. So if the annotation type is `caption`, then it should only have one of them. We refined the tabular annotation like if "Title" is "It's really good!", we transformed it to "Title:It's really good! We have not yet covered other normal image datasets that are not refined tabular datasets. So if the annotation type is caption, then it should be singular. To cover other normal image dataset, we will revisit this.

Reply:
That seems an implicit condition in Datumaro. What I want is to make everything explicitly if possible. Relying on the implicit condition can lead to a bug. If someone develops something that can changes the implicit condition, he must never be noticed what he is doing can affect to the behavior of this code. It leads to a bad result in software maintenance. Add something like https://github.com/openvinotoolkit/datumaro/pull/1533#discussion_r1651887322

Reply:
That's why I added https://github.com/sooahleex/datumaro/blob/f6d5c3902d59e7f2d84ad8ef74edf3dfad2c9da6/src/datumaro/plugins/transforms.py#L1979-L1982. If this is not enough to show the implicit condition, please let me know. I will consider it.